### PR TITLE
Implemented all remaining swagger api calls, implemented contract interface (still unstable)

### PR DESCRIPTION
**Pull request by devsnd**
Implemented all of the swagger interface calls and also started implementing the contract interface. All those calls have at least one test but some of those tests still fail because the underlying interface is not documented enough to make them work.

However I like having failing tests better than none, this allows us to track progress on this interface.

### CODE REVIEW

File: aeternity/epoch.py [hunk 1]
```diff
@@ -178,6 +225,175 @@ def run(self):
...
             return
 
+    #
+    # API functions
+    #
+
+    def get_pubkey(self):
+        return self._config.get_pubkey()
+
+    def get_current_height(self):
```
Comment (tillkolter) [inline aeternity/epoch.py]: In the JS sdk its' called `getBlockHeight`. Do we need to specify that it's `current`? On the other hand, do I need to specify that its `Block`? Would `get_height` probably be enough? :)


... ok, I just checked web3 and the call it `get_height` there. Should we go with that?

File: aeternity/epoch.py [hunk 2]
```diff
@@ -178,6 +225,175 @@ def run(self):
...
+            offset=None,
+            limit=None,
+            tx_types=None,
+            exclude_tx_types=None,
+    ):
+        """
+        get the transactions of the account `account_pubkey`. If left empty, gets
+        the transactions of this node's account.
+        """
+        if account_pubkey is None:
```
Comment (tillkolter) [inline aeternity/epoch.py]: good switch. i will follow your lead for that with my implementation

File: aeternity/epoch.py [hunk 3]
```diff
@@ -43,6 +42,59 @@ class EpochRequestError(Exception):
...
+    'last_30_blocks_time'
+])
+LastBlockInfo = namedtuple('BlockInfo', [
+    'difficulty', 'height', 'time', 'time_delta_to_parent'
+])
+BlockWithTx = namedtuple('BlockWithTx', [
+    'data_schema', 'hash', 'height', 'nonce', 'pow', 'prev_hash', 'state_hash',
+    'target', 'time', 'transactions', 'txs_hash', 'version'
+])
+
```
Comment (tillkolter) [inline aeternity/epoch.py]: Good point to have I/O validation. I will see how I can port that to the JS world.

File: aeternity/epoch.py [hunk 4]
```diff
@@ -178,6 +225,175 @@ def run(self):
...
+        """
+        assert not (height is not None and hash is not None), "Can only set either height or hash!"
+        if account_pubkey is None:
+            account_pubkey = self.get_pubkey()
+        params = {}
+        if height is not None:
+            params['height'] = height
+        if hash is not None:
+            params['hash'] = hash
+        return self.internal_http_get(f'account/balance/{account_pubkey}', params=params)['balance']
```
Comment (tillkolter) [inline aeternity/epoch.py]: What happens when both parameters are set and conflict with each other?

File: aeternity/epoch.py [hunk 5]
```diff
@@ -178,6 +225,175 @@ def run(self):
...
+            params['limit'] = limit
+        params.update(self.make_tx_types_params(tx_types, exclude_tx_types))
+
+        resp = self.internal_http_get(f'account/txs/{account_pubkey}', params=params)
+        return [
+            transaction_from_dict(transaction) for transaction in resp['transactions']
+        ]
+
+    def get_all_balances(self):
+        # TODO: how does this call make any sense?? all balanaces?! --devsnd
```
Comment (tillkolter) [inline aeternity/epoch.py]: 👍

File: aeternity/epoch.py [hunk 6]
```diff
@@ -178,6 +225,175 @@ def run(self):
...
+        # TODO
+        raise NotImplementedError()
+
+    def get_block_by_height(self, height):
+        empty_block = {  # this data does not exist in the genesis block struct
+            'pow': None, 'prev_hash': None, 'transactions': [],
+            'txs_hash': None, 'data_schema': None, 'hash': None
+        }
+        data = self.local_http_get('block-by-height', params=dict(height=height))
+        print(empty_block)
```
Comment (tillkolter) [inline aeternity/epoch.py]: better remove print statement or make it a `logger`

File: aeternity/epoch.py (hunk 1 above)
Comment (devsnd) [reply aeternity/epoch.py]: yeah, `get_height` it is. that will make it easier for people coming from web3

File: aeternity/epoch.py (hunk 4 above)
Comment (devsnd) [reply aeternity/epoch.py]: This is handled 4 lines above in the `assert` statement already

**Pull request merged**